DataFormats/SoATemplate/test/SoALayoutAndView_t.hip.cc

```diff
@@ -123,6 +125,12 @@ int main(void) {
   hipCheck(hipMallocHost((void**)&h_buf, hostDeviceSize));
   SoAHostDeviceLayout h_soahdLayout(h_buf, numElements);
   SoAHostDeviceView h_soahd(h_soahdLayout);
+
+  // Validation of range checking variants initialization
+  SoAHostDeviceRangeCheckingView h_soahdrc(h_soahdLayout);
+  [[maybe_unused]] SoAHostDeviceRangeCheckingView h_soahdrc2 = h_soahdLayout;
+  [[maybe_unused]] SoAHostDeviceRangeCheckingView h_soahdrc3{h_soahd};
+  [[maybe_unused]] SoAHostDeviceRangeCheckingView h_soahdrc4 = h_soahd;
```

Comment:
I'd suggest to add some tests on the left-hand-side objects as well to ensure the behavior of the "copy" constructor is correct.

Reply:
Sure!